### PR TITLE
[enterprise-4.0] Build Inputs - Concept and Procedures 

### PR DESCRIPTION
**Pull request by bmcelvee**
This is my first attempt to add concept and procedure modules to an assembly (hopefully a step in the right direction :smile_cat:). It's still super rough, and I'm leaving in some of the template text until I'm ready to remove it. 

For what is here so far, I've pulled content from our existing documentation and rewritten/made some edits to make it more concise for the concept.  

@openshift/team-documentation FYI

### CODE REVIEW

File: modules/define-build-inputs.adoc [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // Module included in the following assemblies:
 //
 // <List assemblies here, each on a new line>
+//<assembly/creating-build-inputs.adoc>
```
Comment (vikram-redhat) [inline modules/define-build-inputs.adoc]: Replace assembly with builds_and_deployments

**Comment by vikram-redhat** [comment]
Looks good to me @bmcelvee!

**Comment by bmcelvee** [comment]
A couple of notes:
* I will need to go back and add links as we add more modules. 
* In the case where two options are provided, I am only documenting the recommended methods. We can address this further with testing. 
** E.g., in the "Source code trusted certificate authorities" module, I left out the not recommended option for creating a secret with a CA cert. file.

**Comment by bmcelvee** [comment]
Closing and opening a new PR to troubleshoot failures.

**Pull request closed without merging**